cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManagerTest.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginKafka;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginKafkaBuilder;
+import io.strimzi.api.kafka.model.kafka.quotas.QuotasPluginStrimzi;
+import io.strimzi.operator.common.AdminClientProvider;
+import io.strimzi.operator.common.Reconciliation;
+import io.vertx.core.Vertx;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AlterClientQuotasResult;
+import org.apache.kafka.clients.admin.DescribeClientQuotasResult;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.quota.ClientQuotaAlteration;
+import org.apache.kafka.common.quota.ClientQuotaEntity;
+import org.apache.kafka.common.quota.ClientQuotaFilter;
+import org.apache.kafka.common.quota.ClientQuotaFilterComponent;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.strimzi.operator.common.auth.TlsPemIdentity.DUMMY_IDENTITY;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class DefaultKafkaQuotasManagerTest {
+
+    private final Long defaultProducerByteRate = 2000L;
+    private final Long defaultConsumerByteRate = 2000L;
+    private final Double defaultMutationRate = 0.5;
+    private final Integer defaultRequestPercentage = 25;
+
+    private static Vertx vertx;
+
+    @BeforeAll
+    public static void before() {
+        vertx = Vertx.vertx();
+    }
+
+    @AfterAll
+    public static void after() {
+        vertx.close();
+    }
+
+    @Test
+    void testReconfigureAllDefaultQuotasSetInKafka(VertxTestContext context) {
+        long consumerByteRate = 1000;
+        long producerByteRate = 1000;
+        double mutationRate = 0.1;
+        int requestPercentage = 33;
+
+        Checkpoint checkpoint = context.checkpoint();
+
+        Map<String, Double> configuredInKafka = new HashMap<>();
```

Comment:
Should you be using Captors instead of this?

Reply:
I didn't know about the Captors, that's a good point, thanks :)